streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -538,13 +539,16 @@ public void flush() {
                 } catch (final RuntimeException exception) {
                     if (firstException == null) {
                         // do NOT wrap the error if it is actually caused by Streams itself
-                        if (exception instanceof StreamsException)
+                        // In case of FailedProcessingException Do not keep the failed processing exception in the stack trace
+                        if (exception instanceof FailedProcessingException)
+                            firstException = new StreamsException(exception.getCause());
+                        else if (exception instanceof StreamsException)
                             firstException = exception;
                         else
                             firstException = new ProcessorStateException(
                                 format("%sFailed to flush state store %s", logPrefix, store.name()), exception);
                     }
-                    log.error("Failed to flush state store {}: ", store.name(), exception);
+                    log.error("Failed to flush state store {}: ", store.name(), exception.getCause());
```

Comment:
Why did you change this? If a `StreamsException` that is NOT a `FailedProcessingException` is thrown, it would be perfectly fine to log the exception.
You could do something like:

```java
log.error("Failed to flush state store {}: ", store.name(), exception instanceof FailedProcessingException ? exception.getCause() : exception);
``` 

or maybe it is cleaner to copy the log message inside the `if (firstException == null) {`.

Reply:
I copied the log.error in the `if (firstException == null)`:
`log.error("Failed to flush cache of store {}: ", store.name(), firstException);`
else 
`log.error("Failed to flush cache of store {}: ", store.name(), exception);`